includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1856,6 +1856,10 @@ public function validate_account_statement_descriptor_field( $key, $value ) {
 	public function add_order_actions( $actions ) {
 		global $theorder;
 
+		if ( ! is_object( $theorder ) ) {
```

Comment:
This can lead to a bug in the future in case the next if is changed or code is added between the ifs. It is probably better to actually validate if the order is an object and in case it is not just return the $actions instead of trying to execute the rest of the code with an invalid or dummy order. This conditional can be added inside the next if as it already returns

Reply:
The code is inspired on the 1031-gh-woocommerce/automatewoo code changes, I think that when another plugin call `woocommerce_order_actions` on WCPayments it wants to call the `capture_charge` and the `cancel_authorization` that is added on the following code even when we do not a global order, what do you think?

Reply:
That PR in AutomateWoo repository is just a patch meant to temporary fix the error without having to wait for a fix on WC Payments. Basically it creates a dummy order just so when the filter `woocommerce_order_actions` is called the function `add_order_actions` has an order to check and return without doing anything. 

As we are fixing WC Payments we don't need the dummy order anymore. We actually want to check if the `$the_order` exists and if it does not just return the $actions for the next function to use. This way they can remove the dummy order in a future PR (which we will need to create an issue for on AutomateWoo). 

> I think that when another plugin call woocommerce_order_actions on WCPayments it wants to call the capture_charge and the cancel_authorization that is added on the following code even when we do not a global order, what do you think?

As it is right now with the dummy order the `capture_charge` and `cancel_authorization` will not be added as the payment method is empty. So basically we are not using the dummy order and returning before removing it. So this could have some side effects.